src/service.ts

```diff
@@ -331,6 +333,34 @@ export class Service {
     }
     return uri;
   }
+
+  static downloadProject(project: Project, uri?: string) {
+    const zipFile: JSZip = new JSZip();
+    let zipName: string = "download.zip";
+    if (!isUndefined(uri)) {
+      zipName = `${uri}.zip`;
+    }
+    function defineZip(file: File, folder?: any) {
+      if (file instanceof Directory) {
+        let folder: any;
+        if (!(file instanceof Project)) {
+          folder = zipFile.folder(file.name);
+        }
+        file.mapEachFile((file: File) => defineZip(file, folder));
+      } else {
+        if (!isUndefined(folder)) {
+          folder.file(file.name, file.data);
+        } else {
+          zipFile.file(file.name, file.data);
+        }
+      }
+    }
+    defineZip(project);
+    zipFile.generateAsync({type: "blob"}).then(function(blob: any) {
```

Comment:
nit: use arrow function instead

Reply:
see also https://stuk.github.io/jszip/documentation/api_jszip/generate_async.html#mimetype-option example how to specify a mime type for generated blob.